dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -7632,98 +7650,16 @@ public void validateContentlet(final Contentlet contentlet, final List<Category>
-                                        field.getVelocityVarName(), contentletSearch.getInode());
-                                Logger.warn(this, errorMsg);
-                                throw new DotContentletValidationException(errorMsg);
-                            }
-                            final Map<String, Object> uniqueContentMap = uniqueContent.getMap();
-                            final Object obj = uniqueContentMap.get(field.getVelocityVarName());
-                            if ((isDataTypeNumber && Objects.equals(fieldValue, obj)) ||
-                                    (!isDataTypeNumber && ((String) obj).equalsIgnoreCase(
-                                            ((String) fieldValue)))) {
-                                unique = false;
-                                break;
-                            }
-
-                        }
-                        if (!unique) {
-                            if (UtilMethods.isSet(contentlet.getIdentifier())) {
-                                Iterator<ContentletSearch> contentletsIter = contentlets.iterator();
-                                while (contentletsIter.hasNext()) {
-                                    ContentletSearch cont = contentletsIter.next();
-                                    if (!contentlet.getIdentifier()
-                                            .equalsIgnoreCase(cont.getIdentifier())) {
-                                        cve.addUniqueField(field);
-                                        hasError = true;
-                                        Logger.warn(this,
-                                                getUniqueFieldErrorMessage(field, fieldValue,
-                                                        cont));
-                                        break;
-                                    }
-                                }
-                            } else {
-                                cve.addUniqueField(field);
-                                hasError = true;
-                                Logger.warn(this, getUniqueFieldErrorMessage(field, fieldValue,
-                                        contentlets.get(0)));
-                                break;
-                            }
-                        }
-                    }
-                } catch (final DotDataException | DotSecurityException e) {
+                try {
+                    UniqueFieldValidationStrategyResolver.INSTANCE.get().validate(contentlet,
+                            LegacyFieldTransformer.from(field));
+                } catch (UniqueFieldValueDuplicatedException e) {
+                    cve.addUniqueField(field);
+                    hasError = true;
+                    Logger.warn(this, getUniqueFieldErrorMessage(field, fieldValue,
+                            UtilMethods.isSet(e.getContentlets()) ? e.getContentlets().get(0) : "UnKnow"));
```

Comment:
Typo. The word should be `Unknown`

Reply:
done https://github.com/dotCMS/core/pull/30466/files#diff-fa1ceaa19618a6b2bbc30e24c6f930b4971f417db50babb748c2e2837ba9eb82R7676